bioconda_utils/lint_functions.py

```diff
@@ -0,0 +1,207 @@
+    """
+    Does the package exist in bioconda?
+    """
+    results = _subset_df(recipe, meta, df)
+    channels = set(results.channel)
+    if 'bioconda' in channels:
+        return {
+            'already_in_bioconda': True,
+            'fix': 'bump version or build number'
+        }
+
+
+def missing_home(recipe, meta, df):
+    try:
+        meta['about']['home']
+    except KeyError:
+        return {
+            'missing_home': True,
+            'fix': 'add about:home',
+        }
+
+
+def missing_summary(recipe, meta, df):
+    try:
+        meta['about']['summary']
+    except KeyError:
+        return {
+            'missing_summary': True,
+            'fix': 'add about:summary',
+        }
+
+
+def missing_license(recipe, meta, df):
+    try:
+        meta['about']['license']
+    except KeyError:
+        return {
+            'missing_license': True,
+            'fix': 'add about:license'
+        }
+
+
+def missing_tests(recipe, meta, df):
+    test_files = ['run_test.py', 'run_test.sh', 'run_test.pl']
+    if 'test' not in meta:
+        if not any([os.path.exists(os.path.join(recipe, f)) for f in test_files]):
+            return {
+                'no_tests': True,
+                'fix': 'add basic tests',
+            }
+
+
+def missing_hash(recipe, meta, df):
+    # could be a meta-package if no source section or if None
+    try:
+        src = meta['source']
+        if src is None:
+            return
+    except KeyError:
+        return
+
+
+    if not any(
+        (
+            'md5' in src,
+            'sha1' in src,
+            'sha256' in src
+        )
+    ):
+        return {
+            'missing_hash': True,
+            'fix': 'add md5, sha1, or sha256 hash to "source" section',
+        }
+
+
+def uses_git_url(recipe, meta, df):
+    try:
+        src = meta.get('source', {})
+        if src is None:
+            # metapackage?
+            return
+
+        if 'git_url' in src:
+            return {
+                'uses_git_url': True,
+                'fix': 'use tarballs whenever possible',
+            }
+    except KeyError:
+        return
+
+
+def _get_deps(meta):
+    reqs = meta.get('requirements')
+    if reqs is None:
+        return []
+    return [
+        i for i in
+        [reqs.get('build'), reqs.get('run'), reqs.get('test')]
+        if i is not None
+    ]
+
+
+def uses_perl_threaded(recipe, meta, df):
+    if 'perl-threaded' in _get_deps(meta):
+        return {
+            'depends_on_perl_threaded': True,
+            'fix': 'use "perl" instead of "perl-threaded"',
+        }
+
+
+def uses_javajdk(recipe, meta, df):
+    if 'java-jdk' in _get_deps(meta):
+        return {
+            'depends_on_java-jdk': True,
+            'fix': 'use "openjdk" instead of "java-jdk"',
+
+               }
+
+def uses_setuptools(recipe, meta, df):
+    if 'setuptools' in _get_deps(meta):
+        return {
+            'depends_on_setuptools': True,
```

Comment:
Right, currently tests fail if any lint is produced, but there is a mechanism for skipping them on a case-by-case basis. But good point here on setuptools as run dependency. I had added this because I remember Bjoern pointing out setuptools in meta.yamls, but I don't remember if this was in the run or build. I'll change it to build.